### PR TITLE
Daniil Golubev Pull Requests

### PR DESCRIPTION
**Pull request by alexthvest**

### CODE REVIEW

**Comment by jskonst** [comment]
Will take a closer look

File: CourseApp/CourseApp/CourseApp.cpp [hunk 1]
```diff
@@ -0,0 +1,63 @@
+#include <iostream>
+#include <math.h>
+#include <vector>
+
+double func(const double x)
+{
+    if (abs(x) < 1)
```
Comment (jskonst) [inline CourseApp/CourseApp/CourseApp.cpp:8]: Please always use parentheses

**Review by jskonst — CHANGES_REQUESTED** [review]

File: CourseApp/CourseApp/CourseApp.cpp [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+        return acos(x);
+
+    return pow(1.2, x) - pow(x, 1.2);
+}
+
+std::vector<double> exampleA(const double xStart, const double xEnd, const double xDelta)
```
Comment (jskonst) [inline CourseApp/CourseApp/CourseApp.cpp]: let's return vector of vectors - that will contain both x and y values (only for TaskA and taskB functions)

File: CourseApp/CourseApp/CourseApp.cpp [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    return results;
+}
+
+std::vector<double> exampleB(const double *values, const int size)
+{
+    std::vector<double> results;
+
+    for (int i = 0; i < size; i++)
```
Comment (jskonst) [inline CourseApp/CourseApp/CourseApp.cpp]: Can we iterate over vector without using indexes?

**Review by jskonst — CHANGES_REQUESTED** [review]

File: CourseApp/CourseApp/CourseApp.cpp [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    for (double x = xStart; x <= xEnd; x += xDelta)
+    {
+        double result = func(x);
+        results.push_back(result);
+    }
+
+    return results;
+}
+
+std::vector<double> exampleB(const double *values, const int size)
```
Comment (jskonst) [inline CourseApp/CourseApp/CourseApp.cpp]: Can we pass vector and get it's size?

**Review by jskonst — APPROVED** [review]

File: CourseApp/CourseApp/CourseApp.cpp (hunk 1 above)
Comment (alexthvest) [reply CourseApp/CourseApp/CourseApp.cpp:8]: ok, i will use ternary :)

File: CourseApp/CourseApp/CourseApp.cpp (hunk 3 above)
Comment (alexthvest) [reply CourseApp/CourseApp/CourseApp.cpp]: for (type item : vector) {}

File: CourseApp/CourseApp/CourseApp.cpp (hunk 4 above)
Comment (alexthvest) [reply CourseApp/CourseApp/CourseApp.cpp]: maybe

**Review by jskonst — APPROVED** [review]

**Pull request merged**